server/paymentProviders/stripe/alipay.ts

```diff
@@ -0,0 +1,214 @@
+      });
+
+      // Create a Transaction
+      const fees = extractFees(balanceTransaction, balanceTransaction.currency);
+      const amountInHostCurrency = convertFromStripeAmount(balanceTransaction.currency, balanceTransaction.amount);
+      const hostFeeInHostCurrency = await getHostFee(amountInHostCurrency, order);
+      const data = {
+        charge,
+        balanceTransaction,
+        isFeesOnTop: order.data?.isFeesOnTop,
+        isSharedRevenue,
+        platformFee: platformFee,
+        platformTip,
+        hostFeeSharePercent,
+      };
+      const hostCurrencyFxRate = amountInHostCurrency / order.totalAmount;
+      const platformFeeInHostCurrency = isSharedRevenue ? platformTip * hostCurrencyFxRate || 0 : fees.applicationFee;
+
+      const transactionPayload = {
+        CreatedByUserId: order.CreatedByUserId,
+        FromCollectiveId: order.FromCollectiveId,
+        CollectiveId: order.CollectiveId,
+        PaymentMethodId: order.PaymentMethodId,
+        type: TransactionTypes.CREDIT,
+        OrderId: order.id,
+        amount: order.totalAmount,
+        currency: order.currency,
+        hostCurrency: balanceTransaction.currency.toUpperCase(),
+        amountInHostCurrency,
+        hostCurrencyFxRate,
+        paymentProcessorFeeInHostCurrency: fees.stripeFee,
+        taxAmount: order.taxAmount,
+        description: order.description,
+        hostFeeInHostCurrency,
+        platformFeeInHostCurrency,
+        data,
+      };
+
+      await models.Transaction.createFromContributionPayload(transactionPayload, {
+        isPlatformTipDirectlyCollected: true,
+      });
+      await order.update({ status: 'PAID' });
+
+      res.redirect(`${config.host.website}/${order.collective.slug}/donate/success?OrderId=${OrderId}`);
+    } else if (redirect_status === 'failed') {
+      const id = idDecode(OrderId, IDENTIFIER_TYPES.ORDER);
+      debug(`payment for order ${id} failed, deleting order`);
+      const order = await models.Order.findByPk(id, {
+        include: [{ model: models.Collective, as: 'collective' }],
+      });
+      if (order) {
+        await order.destroy();
+        res.redirect(
+          `${config.host.website}/${order.collective.slug}/donate?${querystring.stringify({
+            error: "Couldn't approve Alipay payment, please try again.",
+          })}`,
+        );
+      } else {
+        next(new Error('Could not find the requested orded.'));
+      }
+    }
+  } catch (e) {
+    logger.error(e);
+    next(e);
+  }
+};
+
+const webhook = async (_, event) => {
+  if (event.type === 'charge.refund.updated') {
+    const refund = event.data.object;
+    if (refund.status === 'succeeded') {
+      const transaction = await models.Transaction.findOne({
+        where: { type: 'CREDIT', isRefund: false, data: { charge: { id: refund.charge } } },
+        include: [{ model: models.Collective, as: 'collective' }, { model: models.PaymentMethod }],
```

Comment:
:ice_cream: There's probably almost 0 chance of a conflict on a charge ID with another provider, but I find it cleaner to be explicit about what we're searching here (and it saved us a `if (transaction.PaymentMethod.type !== 'alipay')` below) 

```suggestion
        include: [{ model: models.Collective, as: 'collective' }, { model: models.PaymentMethod, require: true, where: { type: 'alipay' } }],
```